core/src/main/java/org/elasticsearch/index/query/HasChildQueryParser.java

```diff
@@ -33,7 +33,12 @@
  */
 public class HasChildQueryParser implements QueryParser<HasChildQueryBuilder> {
 
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField TYPE_FIELD = new ParseField("child_type", "type");
```

Comment:
From the docs it seems that "type" is the preferred name here, can't judge this from the code. Can you re-check?
https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-has-child-query.html